core/src/main/java/hudson/util/StreamResource.java

```diff
@@ -39,12 +40,32 @@
      *      Used for display purpose.
      */
     public StreamResource(String name, InputStream in) {
```

Comment:
Is it intentionally not annotated with `@NonNull`?  It seems that if it must be non-null, then the annotation might help some of the static analysis utilities detect null arguments being passed.
```suggestion
    public StreamResource(String name, @NonNull InputStream in) {
```

Reply:
Good point. I added the annotation.